src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx

```diff
@@ -232,33 +242,108 @@ function LinkFilesTab() {
     return manualLinkFileRows;
   }, [selectedRows]);
 
-  const makeLinks = async () => {
-    await Promise.all(map(groupedLinksMap, async (fileIds, episodeID) => {
-      const payload: FileLinkApiType = {
-        episodeID: toNumber(episodeID),
-        fileIDs: [],
-      };
-      payload.episodeID = toNumber(episodeID);
-      forEach(fileIds, (link) => {
-        if (link.FileID === 0) {
-          return;
+  useEffect(() => {
+    if (!isLinking || isLinkingRunning || getAnidbSeriesQuery.isLoading || !(getAnidbSeriesQuery.isSuccess && getAnidbSeriesQuery.data.ShokoID)) {
+      return;
+    }
+
+    const seriesID = getAnidbSeriesQuery.data.ShokoID;
+    const linkFiles = async () => {
```

Comment:
seems to me that linkFiles does not have to be inside the useEffect it only makes it look a lot more complicated than it should be